std/parallelism.d

```diff
@@ -99,86 +99,56 @@ else version(NetBSD)
     version = useSysctlbyname;
 }
 
-
-version(Windows)
-{
-    // BUGS:  Only works on Windows 2000 and above.
-    shared static this()
-    {
-        import core.sys.windows.windows : SYSTEM_INFO, GetSystemInfo;
-        import std.algorithm.comparison : max;
-
-        SYSTEM_INFO si;
-        GetSystemInfo(&si);
-        totalCPUs = max(1, cast(uint) si.dwNumberOfProcessors);
-    }
-
-}
-else version(linux)
-{
-    shared static this()
-    {
-        import core.sys.posix.unistd : _SC_NPROCESSORS_ONLN, sysconf;
-        totalCPUs = cast(uint) sysconf(_SC_NPROCESSORS_ONLN);
-    }
-}
-else version(Solaris)
+/*
+A lazily initialized global constant. The underlying value is a shared global
+statically initialized to `outOfBandValue` which must not be a legit value of
+the constant. Upon the first call the situation is detected and the global is
+initialized by calling `initializer`. The initializer is assumed to be pure
+(even if not marked as such), i.e. return the same value upon repeated calls.
+For that reason, no special precautions are taken so `initializer` may be called
+more than one time leading to benign races on the cached value.
+
+In the quiescent state the cost of the function is an atomic load from a global.
+*/
+package @property @nogc nothrow pure @trusted
+T lazilyInitializedConstant(T, T outOfBandValue, alias initializer)()
 {
-    shared static this()
-    {
-        import core.sys.posix.unistd : _SC_NPROCESSORS_ONLN, sysconf;
-        totalCPUs = cast(uint) sysconf(_SC_NPROCESSORS_ONLN);
-    }
+    static nothrow T impl()
+    {
+        static shared T result = outOfBandValue;
+        // Short path
+        auto local = atomicLoad(result);
```

Comment:
should `local` be `const` ?

Reply:
no, for generality

Reply:
I can't think of a case where `const` would pose a problem.